xbmc/windowing/X11/GLContextEGL.cpp

```diff
@@ -58,154 +60,204 @@ bool CGLContextEGL::Refresh(bool force, int screen, Window glWindow, bool &newCo
     return true;
   }
 
-  // create context
-  bool retVal = false;
+  Destroy();
+  newContext = true;
 
-  if (m_eglDisplay == EGL_NO_DISPLAY)
+  if (eglGetPlatformDisplayEXT)
   {
-    m_eglDisplay = eglGetDisplay((EGLNativeDisplayType)m_dpy);
-    if (m_eglDisplay == EGL_NO_DISPLAY)
+    EGLint attribs[] =
     {
-      CLog::Log(LOGERROR, "failed to get egl display\n");
-      return false;
-    }
-    if (!eglInitialize(m_eglDisplay, NULL, NULL))
-    {
-      CLog::Log(LOGERROR, "failed to initialize egl display\n");
-      return false;
-    }
+      EGL_PLATFORM_X11_SCREEN_KHR, screen,
+      EGL_NONE
+    };
+    m_eglDisplay = eglGetPlatformDisplayEXT(EGL_PLATFORM_X11_KHR,(EGLNativeDisplayType)m_dpy,
```

Comment:
Nit: If you use `EGL_EXT_platform_base`, it would make more sense to use `EGL_PLATFORM_X11_EXT` for consistency